airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -219,12 +219,10 @@ def parse_s3_url(s3url: str) -> tuple[str, str]:
         elif format[0] == "https:":
             temp_split = format[1].split(".")
             if temp_split[0] == "s3":
-                split_url = format[1].split("/")
-                bucket_name = split_url[1]
-                key = "/".join(split_url[2:])
+                _, bucket_name, key = format[1].split("/", 2)
```

Comment:
Wouldn’t this be just a subset of `temp_split`? I have a feeling this entire block can be rewritten a bit to split less

Reply:
Had to revert this. In one test, there was a dot in the filename.